src/modules/configs/config.py

```diff
@@ -0,0 +1,48 @@
+import os
+from json import loads
+from typing import Union
+
+from jsonschema import validate, ValidationError
+
+from ingestation.modules.global_vars import APPLICATION_DIR
+
+
+def validate_json(conf: dict, schema_file) -> None:
+    validate(instance=conf, schema=getattr(schema_file, "schema"))
+
+
+class Conf:
+    """Class for parsing configuration files"""
+
+    def __init__(self, conf_path: Union[bool, str], conf_type: str, schema_file):
+        self.conf_type = conf_type
+        self.conf_path = self.get_conf_path(conf_path)
+        self.schema_file = schema_file
+
+    def parse_conf(self) -> Union[bool, dict]:
+        conf = self.load_conf()
+
+        try:
+            validate_json(conf, self.schema_file)
+        except ValidationError as ex:
+            raise ValidationError(
+                f"[CRITICAL]: {self.conf_type.capitalize()} config validation failed.") \
+                from ex
+
+        return conf
+
+    def load_conf(self) -> dict:
+        with open(self.conf_path, "r", encoding="utf-8") as fh:
+            return loads(fh.read())
+
+    def get_conf_path(self, conf_path: Union[str, bool]) -> str:
+        if not conf_path:
+            conf_path = os.path.join(APPLICATION_DIR, f"{self.conf_type}.json")
+            if os.path.isfile(conf_path):
+                return conf_path
+            raise FileNotFoundError(f"[CRITICAL]: {self.conf_type.capitalize()}"
+                                    " configuration could not be obtained.")
```

Comment:
*... could not be obtained/found at {os.path.dirname(self.conf_path)}*

Reply:
Remember to add whitespace between a path and fullstop - nice gesture towards terminal users (double-clicking in Putty marks surrounding interpunction).